### PR TITLE
521 test robustness pgbouncer

### PR DESCRIPTION
**Pull request by rafatower**
This PR makes it possible to execute tests with and without pgbouncer.

This fixes the issue https://github.com/CartoDB/CartoDB-SQL-API/issues/521

### CODE REVIEW

**Comment by rafatower** [comment]
I understand that for this to be really more automated and useful, I'd need to tweak the travis build and docker image to make use of `pgbouncer`, another dependency, and that it might not be desirable.

No big deal and no strong opinion, it is simply that it helped me to run existing tests, taking into account subtle differences between direct vs pgbounced connections.

File: test/support/db_utils.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+    const client = new Client({
+        database: 'pgbouncer',
+        user: dbConfig.db_user,
+        host: dbConfig.db_host,
+        port: dbConfig.db_port
+    });
+
+    // We just chain a PAUSE followed by a RESUME
```
Comment (oleurud) [inline test/support/db_utils.js]: Ei, what is the objective of pause - resume?

Comment (rafatower) [reply test/support/db_utils.js]: https://pgbouncer.github.io/usage.html#admin-console

> PAUSE [db];
> PgBouncer tries to disconnect from all servers, first waiting for all queries to complete. The command will not return before all queries are finished. To be used at the time of database restart.

> RESUME [db];
> Resume work from previous PAUSE or SUSPEND command.

Wait for all the running queries to finish and let the new ones reconnect. This way you pick the changed session parameters such as `set statement_timeout`. Sames happens in production, BTW, but there the connections are refreshed usually more often.

Comment (oleurud) [reply test/support/db_utils.js]: Ok, thanks! :)

File: test/support/db_utils.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    client.connect();
+    client.query('PAUSE', (err, res) => {
```
Comment (dgaubert) [inline test/support/db_utils.js]: Please, follow Node.js conventions:

```js
client.query('PAUSE', (err, res) => {
  if (err) {
    return callback(err);
  }
  client.query('RESUME', (err, res) => ...);
}
```
If  you inmediatly return after calling the callback, there is no need to wrap following sentences into a `else` statement.

File: test/support/db_utils.js (hunk 1 above)
Comment (dgaubert) [inline test/support/db_utils.js]: What about `// We just chain a PAUSE followed by a RESUME to reset internal pool connections of PgBouncer`?

**Comment by rafatower** [comment]
I covered your comments (that I always appreciate) and pleased jshint. Now time to merge. Thanks!

**Comment by oleurud** [comment]
:+1:

**Pull request merged**